### DISCUSSION TITLE
How to bind multiple functions using add_key_binding

### DISCUSSION THREAD

**Discussion by ahaoboy** [mpv-player/mpv#14292] [Q&A]
For the demo script, how can I print 1 and 2 on mouse click, I tried different flags, but it seems to only output 2.
```js
mp.add_forced_key_binding(
  "MOUSE_BTN0",
  "MPV_EASY_SPEED",
  function cb(event) {
    print(1)
  },
  {
    // complex: true,
    repeatable: true,
    // forced: false,
    forced: true,
  }
)


mp.add_forced_key_binding(
  "MOUSE_BTN0",
  "MPV_EASY_SPEED2",
  function cb(event) {
    print(2)
  },
  {
    // complex: true,
    // repeatable: true,
    forced: true,
  }
)
```

**Comment by avih** [comment]
Like the [docs](https://mpv.io/manual/master/#lua-scripting-[,flags]]\)) say:
> Trying to map multiple commands to a key will essentially prefer a random binding, while the other bindings are not called.


So the answer is that you can't do that using multiple calls to `mp.add_[forced_]key_binding`.

You can, however, add one binding, where your callback function calls several different functions.

**Comment by ahaoboy** [comment]
Yes, I noticed that the js implementation does not use an array or set to save the callback function, but directly uses a field, which will cause the newly bound function to replace the old function.

[function add_binding(forced, key, name, fn, opts) {](https://github.com/mpv-player/mpv/blob/e7630722d81c1cd36042d0b1368407a92e488742/player/javascript/defaults.js#L310-L328)

**Comment by avih** [comment]
It's not an implementation issue. It's by design, and it's the same for lua and js.
